background/get-addon-settings.js

```diff
@@ -36,7 +36,28 @@ const updatePresetIfMatching = (settings, version, oldPreset = null, preset = nu
   }
 };
 
-chrome.storage.sync.get(["addonSettings", "addonsEnabled"], ({ addonSettings = {}, addonsEnabled = {} }) => {
+async function transitionToNewStorageKeys(addonSettings) {
+  // This will not delete the addonSettings storage item
+  chrome.storage.sync.set(
+    {
+      ...minifySettings(addonSettings, null),
+    },
```

Comment:
Self TODO: it might be a better idea to rename addonSettings to oldAddonSettings.

Contributors might often jump between PRs that are still in old versions. I still need to have some thought on the best way to solve that. Keeping addonSettings intact does not help, as you end up with 2 parallel storages for settings depending on the branch you're on.

Reply:
Moving this discussion to the thread instead of this comment.